core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -164,7 +164,9 @@ class HadoopMapReduceCommitProtocol(
   }
 
   override def commitJob(jobContext: JobContext, taskCommits: Seq[TaskCommitMessage]): Unit = {
-    committer.commitJob(jobContext)
+    if (!dynamicPartitionOverwrite) {
+      committer.commitJob(jobContext)
+    }
```

Comment:
Thank you for making a PR. Can we have a test case for this, @turboFei ?
cc @wangyum

Reply:
done.

Reply:
What if we set `spark.hadoop.mapreduce.fileoutputcommitter.algorithm.version` to 2?

Reply:
It works well  when we set spark.hadoop.mapreduce.fileoutputcommitter.algorithm.version to 2.

But for the version 2, it may produce partial result when we kill an application, which is committing tasks.